services/common/src/components/forms/RenderLargeSelect.tsx

```diff
@@ -1,86 +1,69 @@
-import React from "react";
-import PropTypes from "prop-types";
+import React, { FC } from "react";
 import { Select, Form } from "antd";
+import { BaseInputProps } from "./BaseInput";
+import { IOption } from "@mds/common/interfaces/common/option.interface";
 
 /**
- * @constant RenderLargeSelect - Ant Design `AutoComplete` component for redux-form -- being used instead of 'RenderSelect' for large data sets that require a limit.
+ * @component RenderLargeSelect - Ant Design `AutoComplete` component for redux-form -- being used instead of 'RenderSelect' for large data sets that require a limit.
  */
+interface LargeSelectProps extends BaseInputProps {
+  dataSource: any[];
+  handleSearch?: (value) => void | Promise<void>;
+  handleSelect?: (value, option) => void | Promise<void>;
+  handleFocus?: () => void | Promise<void>;
+  selectedOption?: IOption
 
-const propTypes = {
-  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
-  label: PropTypes.string,
-  placeholder: PropTypes.string,
-  dataSource: PropTypes.arrayOf(PropTypes.any).isRequired,
-  selectedOption: PropTypes.shape({
-    key: PropTypes.string,
-    label: PropTypes.string, // was PropTypes.label
-    value: PropTypes.string,
-  }).isRequired,
-  input: PropTypes.shape({
-    value: PropTypes.string,
-    onChange: PropTypes.func,
-    onFocus: PropTypes.func,
-  }).isRequired,
-  meta: PropTypes.shape({
-    touched: PropTypes.bool,
-    error: PropTypes.string,
-    warning: PropTypes.string,
-  }).isRequired,
-  handleSearch: PropTypes.func,
-  handleSelect: PropTypes.func,
-  handleFocus: PropTypes.func,
-  disabled: PropTypes.bool,
-  required: PropTypes.bool,
-};
+}
 
-const doNothing = () => { };
-const defaultProps = {
-  label: "",
-  placeholder: "",
-  handleSelect: doNothing,
-  handleSearch: doNothing,
-  handleFocus: doNothing,
-  disabled: false,
+const RenderLargeSelect: FC<LargeSelectProps> = ({
+  handleFocus = () => { },
+  handleSearch,
+  handleSelect,
+  dataSource = [],
+  ...props
+}) => {
+  const selectedInData = dataSource.includes((opt) => props.selectedOption?.value === opt.value);
+  let data = dataSource;
```

Comment:
Does this need to be a `let`?  `unshift` should work on a const, no?

Is this that weird part of the `RenderLargeSelect` you mentioned.  I was curious what you had come up against there.  It did seem to have some odd behaviour.

Reply:
Yeah! This is the weird part. Basically took out the auto-search thing in PartySelectField and manually inserted the option into render large select instead (at the top, where, IMO, it makes the most sense to be. But if it was already somewhere in the middle of the list then I wasn't going to fuss with it).

Totally can be a const!